components/script/dom/bindings/str.rs

```diff
@@ -302,6 +308,31 @@ impl DOMString {
     pub fn is_valid_week_string(&self) -> bool {
         parse_week_string(&*self.0).is_ok()
     }
+
+    /// A valid local date and time string should be {date}T{time}
+    /// where date and time are both valid
+    /// https://html.spec.whatwg.org/multipage/#valid-local-date-and-time-string
+    pub fn is_valid_local_date_and_time_string(&self) -> bool {
+        parse_local_date_and_time_string(&*self.0).is_ok()
+    }
+
+    /// A valid normalized local date and time string should be {date}T{time}
+    /// where date and time are both valid, and the time string must be as short as possible
+    /// https://html.spec.whatwg.org/multipage/#valid-normalised-local-date-and-time-string
+    pub fn convert_valid_normalized_local_date_and_time_string(&mut self) {
+        let ((year, month, day), (hour, minute, second)) = parse_local_date_and_time_string(&*self.0).unwrap();
+        let year_str = format_zero(year, 4);
```

Comment:
Isn't this just `format!("{:04}", year)`? [Documentation](https://doc.rust-lang.org/std/fmt/)

Reply:
cool! thanks